### PR TITLE
Add lawnchair@0.6.4 w/ git auto-update

### LINKED ISSUE #8058: [Request] Add lawnchair
**Library name:** lawnchair
**Git repository url:** https://github.com/brianleroux/lawnchair
https://www.npmjs.com/package/lawnchair
**Official homepage:** http://brian.io/lawnchair/

PS: You may need to manually download add all the versions if there is no pre-built files in repo or npm pacakge, at least they are on the website.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/34592349-request-add-lawnchair?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by clairetsai818**
Pull request for issue: #8058 
Related issue(s): #9414 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/brianleroux/lawnchair
 * Official website (optional, not the repository):
 * NPM package url (optional): https://www.npmjs.com/package/lawnchair
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 2123
   - Count of watchers: 78
   - Count of forks: 266
 * NPM download stats (optional):
   - Downloads in the last day: 2
   - Downloads in the last week: 19
   - Downloads in the last month: 102

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

close #8058, cc #9414 @MarkMYoung

### CODE REVIEW

**Comment by clairetsai818** [comment]
@PeterDaveHello Please help me review it, thanks.

File: ajax/libs/lawnchair/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "name": "lawnchair",
+  "license": "MIT",
+  "filename": "lawnchair.min.js",
+  "keywords": [
+    "lawnchair",
+    "lawnchairjs",
+    "lawnchair.js",
+    "Gruntfile",
+    "Gruntfile.js",
+    "Grunfilejs"
```
Comment (PeterDaveHello) [inline ajax/libs/lawnchair/package.json]: Grunfilejs would be meaningless here, it's a tool to control the build process of javascript that this lib uses it, besides this problem, your 6 keywords I would say it's only 2 are valid, others are duplicated form of them.

Comment (PeterDaveHello) [inline ajax/libs/lawnchair/package.json]: I'll suggest you:
- lightweight
- json
- document

File: ajax/libs/lawnchair/package.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/brianleroux/lawnchair.git",
+    "fileMap": [
+      {
+        "basePath": "lib",
+        "files": [
+          "**/*"
```
Comment (PeterDaveHello) [inline ajax/libs/lawnchair/package.json]: It's not a good idea to use wildcard `**/*` in non-`dist`/`build`/`release` type of directory, let's write it's rule.

**Comment by clairetsai818** [comment]
@PeterDaveHello Please help me review it, thanks.

**Comment by PeterDaveHello** [comment]
Let's add homepage info in package.json, e.g. `http://brian.io/lawnchair/`

**Comment by clairetsai818** [comment]
＠PeterDaveHello I've added the homepage info to package.json, please help me review this. Thanks!

**Comment by PeterDaveHello** [comment]
@clairetsai818  70b473e should not be here, please try `git rebase master lawnchair`

**Comment by clairetsai818** [comment]
@PeterDaveHello I've tried `git rebase master lawnchair`, please help me review this. Thanks!

**Comment by PeterDaveHello** [comment]
@clairetsai818 I see nothing here, have you pushed it? You commit should come after my comment!

**Comment by clairetsai818** [comment]
@PeterDaveHello I've tried git rebase master lawnchair, please help me review this. Thanks!

File: ajax/libs/lawnchair/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "lightweight",
+    "json",
+    "document"
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/brianleroux/lawnchair.git",
+    "fileMap": [
+      {
+        "basePath": "lib",
```
Comment (PeterDaveHello) [inline ajax/libs/lawnchair/package.json]: I didn't see a `lib` directory inside v0.6.4, anything wrong?
https://github.com/brianleroux/lawnchair/tree/0.6.4

**Comment by clairetsai818** [comment]
@PeterDaveHello I've revised package.json, please help me review this. Thanks!

**Comment by PeterDaveHello** [comment]
@clairetsai818 you better tell be what's going on, but not only you modified it.

According to its Makefile https://github.com/brianleroux/lawnchair/blob/0.6.4/makefile#L12-L41, take only `src/Lawnchair.js` make not be a good idea, we should confirm with its author if it's possible to provide distribution/release files.

@cdnjs/intern2 please take a look at this case and help discuss with its author.

**Comment by PeterDaveHello** [comment]
ping @kennynaoh

**Comment by MarkMYoung** [comment]
If I understand the discussion, no Lawnchair.js by itself will not work.  There are optional plugins you can include and you must provide at least one adapter compatible with your environment.

Let me know if there is some way I can help.

**Comment by PeterDaveHello** [comment]
Thanks @MarkMYoung, we'll work on it soon 😄

**Comment by PeterDaveHello** [comment]
Ping @clairetsai818

**Comment by clairetsai818** [comment]
@PeterDaveHello I am checking if my PR is ready to be reviewed.

**Comment by PeterDaveHello** [comment]
`/w` -> `w/`

**Comment by clairetsai818** [comment]
@PeterDaveHello @maruilian11 I revised the title please help me review this PR. Thanks!

**Comment by PeterDaveHello** [comment]
I didn't see any updates here ...

**Comment by clairetsai818** [comment]
@PeterDaveHello I rebased this branch, please help me review this, thanks!

**Comment by maruilian11** [comment]
your commit msg is still wrong... check it carefully...

File: ajax/libs/lawnchair/package.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    "url": "https://github.com/brianleroux/lawnchair.git"
+  },
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/brianleroux/lawnchair.git",
+    "fileMap": [
+      {
+        "basePath": "src",
+        "files": [
+          "**/*"
```
Comment (PeterDaveHello) [inline ajax/libs/lawnchair/package.json]: It's not a good idea to use `**/*` under src, I think we discussed about this for several times.

**Comment by PeterDaveHello** [comment]
It's almost been 3 months on this PR, can we push harder on this one and get it merged ASAP?

**Comment by clairetsai818** [comment]
@PeterDaveHello Please help me review this PR.

**Comment by PeterDaveHello** [comment]
@clairetsai818 why didn't I see you asking for review right after the last PR update? That's not how we discussed.

**Comment by clairetsai818** [comment]
@PeterDaveHello I saw you just assign this PR to @pvnr0082t , so I think it's better to ask for review after he finish reviewing this PR.

**Comment by PeterDaveHello** [comment]
@clairetsai818 if you do have reason to rebase the branch, you should also ask for review again, the assign is another thing.

**Comment by clairetsai818** [comment]
@PeterDaveHello Got it, I will do that next time.

**Comment by PeterDaveHello** [comment]
For both v0.6.4, compare to its npm package `lib/lawnchair.js` is 17KB but the one you fetched from GitHub is only 5.5K, what do you think about this? Maybe the files you grabbed from `src` need to be bundled or some other processing, I'm not sure, but I'm sure that you are not sure, am I right? So could you please confirm that? 

As I mentioned in the origin issue, there are release files on the website, although it's only up to v0.6.1, but you can still compare the difference from it, maybe the file on website is the same as the file in npm package, I don't know if there is any reason that makes you ignore it or something like that, but you don't have questions about that so it looks like you do nothing to my notes which can help you do the right thing in the right way.

@clairetsai818 as a student under training, I'll need you to be more careful and serious, not just try to hang on your homework, let's improve that.

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by clairetsai818** [comment]
@PeterDaveHello It is waiting for respons

https://github.com/brianleroux/lawnchair/issues/230

**Comment by clairetsai818** [comment]
There are no response to the issue.

**Comment by extend1994** [comment]
Include the files under `src`. Then the lib can be used in front-end well. I tested it and succeed.

**Comment by PeterDaveHello** [comment]
Ask if @clairetsai818 would like to get it done, if not then you can close it.

**Comment by extend1994** [comment]
@clairetsai818 Would you like to finish this pull request?
If there is no response in next 7 days, I will close it temporarily.

**Comment by clairetsai818** [comment]
@extend1994 Sure. I'll finish this PR next week.

**Comment by sufuf3** [comment]
@clairetsai818 
thanks for contribution.
Would you mind working on this and helping improve this PR?

Many thanks.

**Comment by sufuf3** [comment]
@clairetsai818 Would you like to finish this pull request?
Thanks

**Comment by sashberd** [comment]
@sufuf3 As I saw the PR is OK. The files under `src` are added. What else need to be done in order to finish this PR?

**Comment by sashberd** [comment]
Moved to #13710

**Pull request closed without merging**